src/libltfs/ltfs_fsops_raw.c

```diff
@@ -453,7 +454,24 @@ int ltfs_fsraw_cleanup_extent(struct dentry *d, struct tc_position err_pos, unsi
 			}
 			else {
                 TAILQ_FOREACH_REVERSE_SAFE(ext, &entry->d->extentlist, extent_struct, list, preventry) {
-					if (err_pos.block <= (ext->start.block + ext->bytecount/blocksize)) {
+					if (ext->start.block && ext->bytecount) {
+						extent_last.partition = ltfs_part_id2num(ext->start.partition, vol);
+						/* Calculate the last block of this extent */
+						extent_last.block = ext->start.block + (ext->bytecount / blocksize);
```

Comment:
@piste-jp-ibm so SA always uses fixed length never variable length right?

Reply:
No, a extent is constructed from one or more 512KB fixed blocks and no or one variable block.